magento/pwa-studio
# Address Book Skeleton

## Description

This PR activates the Address Book link in the user account menu to an Address Book skeleton page 🦴 .

It:
  * Navigates the user to the `/account-menu` route
  * Displays current customer addresses if present
  * Displays an "Add an Address" call to action button always (not hooked up)
  * Redirects the user to the homepage if they sign out or if their session expires (navigation happens on next action)

## Related Issue





Closes PWA-187.   🕵️ 🐕  

## Acceptance 



### Verification Stakeholders


@schensley 

### Specification



### Verification Steps


1. Sign in
1. Click the "Address Book" link in the site header user account menu
1. Verify that you are sent to the Address Book page
1. Verify that if you have a saved address (you can add one on the checkout page), it appears
1. Verify that an "Add an Address" button always appears
1. Sign out
1. Verify that you are sent to the homepage
1. Sign in
1. Click the "Address Book" link in the left drawer
1. Verify that you are sent to the Address Book page
1. Simulate your token expiring by modifying the `M2_VENIA_BROWSER_PERSISTENCE__signin_token` localStorage entry and setting its `timeStored` to `0`.
1. Refresh the page
1. See that you are redirected to the home page

## Screenshots / Screen Captures (if appropriate)

### Desktop

![Screen Shot 2020-08-25 at 12 51 44 PM](https://user-images.githubusercontent.com/13182778/91209761-d4257780-e6d1-11ea-9346-6e0e161664d9.png)

### Mobile

![Screen Shot 2020-08-25 at 12 52 04 PM](https://user-images.githubusercontent.com/13182778/91209765-d4be0e00-e6d1-11ea-8dd6-455e121d7516.png)

## Checklist

* I have updated the documentation accordingly, if necessary.
